Riot/Modules/Common/Recents/Views/RecentTableViewCell.m

```diff
@@ -128,18 +138,59 @@ - (void)render:(MXKCellData *)cellData
                                             roomId:roomCellData.roomIdentifier
                                        displayName:roomCellData.roomDisplayname
                                       mediaManager:roomCellData.mxSession.mediaManager];
-        
-        // Presence indicator
-        self.presenceIndicatorView.borderColor = ThemeService.shared.theme.backgroundColor;
-        self.presenceIndicatorView.presence = roomCellData.presence;
-        self.presenceIndicatorView.hidden = roomCellData.presence == MXPresenceUnknown;
+
+        if (!mxDirectUserPresenceObserver && roomCellData.directUserId)
+        {
+            // Observe contact presence change
+            MXWeakify(self);
+            mxDirectUserPresenceObserver = [[NSNotificationCenter defaultCenter] addObserverForName:kMXKContactManagerMatrixUserPresenceChangeNotification object:nil queue:[NSOperationQueue mainQueue] usingBlock:^(NSNotification *notif) {
```

Comment:
Would it make sense to pass the `directUserId` in as the `object` to reduce the number of times this block will get called? As a bonus, if the ID is nil, the observer wouldn't even need to be created.

Reply:
Ah that's a shame, but also seems obvious now you mention it 🤦‍♂️